### PR TITLE
gh-64658: Expand Argument Clinic return converter docs

### LINKED ISSUE #64658: [doc] No Argument Clinic documentation on how to specify a return converter
BPO | [20459](https://bugs.python.org/issue20459)
--- | :---
Nosy | @larryhastings, @furkanonder

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = 'https://github.com/larryhastings'
closed_at = None
created_at = <Date 2014-01-31.16:21:42.779>
labels = ['easy', '3.9', '3.10', '3.11', 'expert-argument-clinic', 'docs']
title = '[doc] No Argument Clinic documentation on how to specify a return converter'
updated_at = <Date 2021-12-06.23:40:05.988>
user = 'https://github.com/brettcannon'
```

bugs.python.org fields:
```python
activity = <Date 2021-12-06.23:40:05.988>
actor = 'iritkatriel'
assignee = 'larry'
closed = False
closed_date = None
closer = None
components = ['Documentation', 'Argument Clinic']
creation = <Date 2014-01-31.16:21:42.779>
creator = 'brett.cannon'
dependencies = []
files = []
hgrepos = []
issue_num = 20459
keywords = ['easy']
message_count = 2.0
messages = ['209803', '364723']
nosy_count = 2.0
nosy_names = ['larry', 'furkanonder']
pr_nums = []
priority = 'normal'
resolution = None
stage = 'needs patch'
status = 'open'
superseder = None
type = None
url = 'https://bugs.python.org/issue20459'
versions = ['Python 3.9', 'Python 3.10', 'Python 3.11']
```

</p></details>

### Linked PRs
* gh-104175

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-64658

----
:books: Documentation preview :books:: https://cpython-previews--104175.org.readthedocs.build/

### CODE REVIEW

File: Doc/howto/clinic.rst [hunk 1]
```diff
@@ -1069,16 +1069,17 @@ Currently Argument Clinic supports only a few return converters:
     int
     unsigned int
     long
-    unsigned int
+    unsigned long
     size_t
     Py_ssize_t
     float
     double
-    DecodeFSDefault
```
Comment (erlend-aasland) [inline Doc/howto/clinic.rst:1077]: FTR, the `DecodeFSDefault` return converter was removed in gh-11152

**Review by AlexWaygood — COMMENTED** [review]
I only understand AC in the vaguest terms, but that is definitely a sentence that makes grammatical sense! :)

**Comment by erlend-aasland** [comment]
> I only understand AC in the vaguest terms, but that is definitely a sentence that makes grammatical sense! :)

Haha, yeah, but is it too vague? ;) I guess it is ok for the AC docs, but I'd not let something that vague slip into the sqlite3 docs.

**Comment by erlend-aasland** [comment]
BTW, I removed the reference to the `init` converter; it's only used internally by AC anyways.

**Comment by erlend-aasland** [comment]
FTR, I'm landing this soon. We can easily fix nitpicks in the docs later (no issue needed).

File: Doc/howto/clinic.rst [hunk 2]
```diff
@@ -1033,19 +1033,36 @@ you're not permitted to use:
 Using a return converter
 ------------------------
 
-By default the impl function Argument Clinic generates for you returns ``PyObject *``.
-But your C function often computes some C type, then converts it into the ``PyObject *``
+By default the impl function Argument Clinic generates for you returns
```
Comment (AlexWaygood) [inline Doc/howto/clinic.rst]:
```suggestion
By default, the impl function Argument Clinic generates for you returns
```

**Review by AlexWaygood — COMMENTED** [review]
It looks fine to me, but I still feel like a very underqualified reviewer here :)

File: Doc/howto/clinic.rst (hunk 2 above)
Comment (erlend-aasland) [reply Doc/howto/clinic.rst]: Thanks!

File: Doc/howto/clinic.rst [hunk 3]
```diff
@@ -1033,19 +1033,36 @@ you're not permitted to use:
 Using a return converter
 ------------------------
 
-By default the impl function Argument Clinic generates for you returns ``PyObject *``.
-But your C function often computes some C type, then converts it into the ``PyObject *``
+By default, the impl function Argument Clinic generates for you returns
+:c:type:`PyObject * <PyObject>`.
+But your C function often computes some C type, then converts it into the
+:c:type:`!PyObject *`
```
Comment (erlend-aasland) [inline Doc/howto/clinic.rst]: Let me wrap this neater, now that we're touching this anyways.

```suggestion
But your C function often computes some C type,
then converts it into the :c:type:`!PyObject *`
```

**Pull request merged**